test/object-store/sync/flx_migration.cpp

```diff
@@ -814,67 +814,51 @@ TEST_CASE("Async open + client reset", "[flx][migration][baas]") {
         std::make_shared<util::StderrLogger>(realm::util::Logger::Level::TEST_LOGGING_LEVEL);
 
     const std::string base_url = get_base_url();
-    const std::string partition = "migration-test";
+    const std::string partition = "async-open-migration-test";
     ObjectSchema shared_object("Object", {{"_id", PropertyType::ObjectId, Property::IsPrimary{true}},
                                           {"string_field", PropertyType::String | PropertyType::Nullable},
                                           {"realm_id", PropertyType::String | PropertyType::Nullable}});
     const Schema mig_schema{shared_object};
     size_t num_before_reset_notifications = 0;
     size_t num_after_reset_notifications = 0;
-    auto server_app_config = minimal_app_config(base_url, "server_migrate_rollback", mig_schema);
+    auto server_app_config = minimal_app_config(base_url, "async_open_during_migration", mig_schema);
+    std::unique_ptr<SyncTestFile> config;
     TestAppSession session(create_app(server_app_config));
-    SyncTestFile config(session.app(), partition, server_app_config.schema);
-    config.sync_config->client_resync_mode = ClientResyncMode::DiscardLocal;
-    config.sync_config->notify_before_client_reset = [&](SharedRealm before) {
+    config = std::make_unique<SyncTestFile>(session.app(), partition, server_app_config.schema);
```

Comment:
I don't see any reason for changing this? The unique_ptr itself is never used anywhere so this is just a more complicated way to do the same thing.

Reply:
I thought that the SyncTestFile should run its destructor (deleting from the file system) after the session had cleaned up and stopped sync. There was no default constructor for the SyncTestFile so I used a unique_ptr to initialize it after the session but still have it destructed later. It turns out that this did not fix the issue though, so I'm still poking around to see what the problem is. Unfortunately, I can't repro this locally.

Reply:
This is a very trivial thing, but `std::optional` is a slightly better way to do this (if it's actually needed?).

Reply:
Done. I still think it is more correct to have `~TestAppSession()` run before `~SyncTestFile`. Although it won't really matter when the tests run through the happy path, it did hide the underlying error I was seeing on CI when it was the other way around and the Realm on the file system was deleted with an outstanding sync session that was still in an active client reset.